### PR TITLE
SMS support for macOS!

### PR DESCRIPTION
**Pull request by rshah713**
Extending support for sending messages for macOS systems!

Couple things to note:
- The implementation is really meant for iMessages, as are natively supported by macOS. SMS will only work on macOS if a device that is carrier-activated is [correctly paired and configured](https://support.apple.com/guide/messages/get-sms-texts-from-iphone-on-your-mac-icht8a28bb9a/mac). If SMS is desired, simply passing it into `sms.send(mode='SMS')` will correctly switch the mode. 
- My implementation has been tested with the existing example, and does work on on `macOS Monterey 12.5`

### CODE REVIEW

**Comment by rshah713** [comment]
Hey @tshirtman @misl6!

Apologies for pinging everyone, this is far from a critical PR but it wasn't clear if anyone had reviewed this in the last month of it being introduced. Was hoping to get some feedback on this PR before it became stale, thanks for your time!

File: plyer/facades/sms.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ class Sms:
     Sms facade.
     '''
 
-    def send(self, recipient, message):
+    def send(self, recipient, message, mode='iMessage', **kwargs):
```
Comment (misl6) [inline plyer/facades/sms.py:36]: `iMessage` is not something that could be supported on Android and iOS.
Better to keep `mode=None` here, and than set the default for macOS somewhere else.

File: plyer/platforms/macosx/sms.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+from subprocess import Popen, PIPE
+from plyer.facades import Sms as SMS
+from plyer.utils import whereis_exe
+
+
+class OSXSMS(SMS):
```
Comment (misl6) [inline plyer/platforms/macosx/sms.py:6]: Minor: `OSXSMS` --> `MacOSSMS` ?

File: plyer/platforms/macosx/sms.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    '''
+
+    def _send(self, **kwargs):
+        '''
+        Will send `message` to `recipient` via Messages app
+
+        Currently only supports iMessage as macOS can send that standalone
+        In order to support SMS, a valid carrier-activated device must be
+        connected & configured to macOS
+            - activate this by passing in mode='SMS'
```
Comment (misl6) [inline plyer/platforms/macosx/sms.py:18]: Is something tested and that we can officially support?

**Review by misl6 — CHANGES_REQUESTED** [review]
Hi @rshah713 !
I've left some comments, let me know what you think about it :)

File: plyer/facades/sms.py (hunk 1 above)
Comment (rshah713) [reply plyer/facades/sms.py:36]: Ok, agreed. `kwargs.get(...)` inside `MacOSSMS` can pass the default value.

File: plyer/platforms/macosx/sms.py (hunk 2 above)
Comment (rshah713) [reply plyer/platforms/macosx/sms.py:6]: Done.

File: plyer/platforms/macosx/sms.py (hunk 3 above)
Comment (rshah713) [reply plyer/platforms/macosx/sms.py:18]: Yep, we can officially support it as long as the phone is correctly paired. You can test it and watch the messages turn green as it sends via SMS.

**Comment by rshah713** [comment]
Hey @misl6. As always, thanks for the in-depth review. Requested changes have been made!

**Comment by misl6** [comment]
HI @rshah713 , can you take care of the `E128` that the CI is showing?

File: plyer/facades/sms.py (hunk 1 above)
Comment (misl6) [reply plyer/facades/sms.py]: Yep, but users will not notice the availability of the `mode` setting.

The suggestion is to keep `mode` set to a default value (but not `iMessage`), and to clarify where and how should be used on the docstring for the `send` method.

Users will refer to the following documentation when writing their apps🙂: 
https://plyer.readthedocs.io/en/latest/#plyer.facades.Sms

Comment (rshah713) [reply plyer/facades/sms.py]: @misl6 Done. Now `mode=None` is always passed into `MacOSSMS` but if its value remains `None` we will set it to `iMessage`.

Have also corrected docstring to reflect this change.

File: plyer/facades/sms.py [hunk 4]
```diff
@@ -33,17 +33,22 @@ class Sms:
...
+    def send(self, recipient, message, mode=None, **kwargs):
         '''
         Send SMS or open SMS interface.
+        Includes optional `mode` parameter for macOS that can be set to
+        `'SMS'` if carrier-activated device is correctly paired and
+        configured to macOS.
 
         :param recipient: The receiver
         :param message: the message
+        :param mode: (optional) macOS only, can be set to 'SMS'
```
Comment (misl6) [inline plyer/facades/sms.py:45]:
```suggestion
        :param mode: (optional, macOS only), can be set to 'iMessage' (default) or 'SMS'
```

File: plyer/platforms/macosx/sms.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
```
Comment (misl6) [inline plyer/platforms/macosx/sms.py:18]:
```suggestion
        By default, if `mode` is not explicitly set, `iMessage` is used.
        In order to use `SMS` mode, a valid carrier-activated device must
        be connected and configured.
```

File: plyer/platforms/macosx/sms.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+        '''
+
+        recipient = kwargs.get('recipient')
+        message = kwargs.get('message')
+        mode = kwargs.get('mode')  # can be SMS
```
Comment (misl6) [inline plyer/platforms/macosx/sms.py:24]:
```suggestion
        mode = kwargs.get('mode')  # Supported modes: iMessage (default), SMS
```

**Review by misl6 — CHANGES_REQUESTED** [review]
Hi @rshah713 !

I've added some suggestions regarding the docstrings. 😀

File: plyer/facades/sms.py (hunk 4 above)
Comment (rshah713) [reply plyer/facades/sms.py]: Done.

File: plyer/platforms/macosx/sms.py (hunk 5 above)
Comment (rshah713) [reply plyer/platforms/macosx/sms.py]: Done.

File: plyer/platforms/macosx/sms.py (hunk 6 above)
Comment (rshah713) [reply plyer/platforms/macosx/sms.py]: Done.

**Comment by rshah713** [comment]
Alright @misl6, all docstring changes have been applied!

**Review by misl6 — APPROVED** [review]
Image 13-12-22 at 18 33

Nice!

LGTM. Thank you!

**Pull request merged**